src/buildercore/core.py

```diff
@@ -248,11 +252,16 @@ def _ec2_connection_params(stackname, username, **kwargs):
     return params
 
 @contextmanager
-def stack_conn(stackname, username=config.DEPLOY_USER, **kwargs):
+def stack_conn(stackname, username=config.DEPLOY_USER, node=None, **kwargs):
     if 'user' in kwargs:
         LOG.warn("found key 'user' in given kwargs - did you mean 'username' ??")
 
-    data = stack_data(stackname, ensure_single_instance=True)[0]
+    data = stack_data(stackname, ensure_single_instance=False)
+    ensure(len(data) == 1 or node, "stack is clustered and no node specified")
```

Comment:
error message of this could include the value of `len(data)` `clustered (3 nodes)`